### DISCUSSION TITLE
Latest Google Chrome deb package not found during the Cypress set up in Docker

### DISCUSSION THREAD

**Discussion by francescoleddafrontiersin** [cypress-io/cypress#27812] [Questions and Help]
Hi everyone,

we incurred in the following error just yesterday

![image](https://github.com/cypress-io/cypress/assets/143416063/8071d78f-52ae-4cc6-af5a-b5adc61957e0)

We didn't touch our code and after investigation we found only this difference

![image](https://github.com/cypress-io/cypress/assets/143416063/ba475b1b-9ef5-41a8-ba87-79e8a3374bed)

In our Dockerfile, we download and install the Google Chrome .deb package and discovered that the new version was released just the day before the error.

So using the latest stable version it was broken: 

`RUN wget https://dl.google.com/linux/direct/google-chrome-stable_current_amd64.deb
RUN dpkg -i google-chrome-stable_current_amd64.deb`

We resolved using a fixed previous version:

`RUN wget https://dl.google.com/linux/chrome/deb/pool/main/g/google-chrome-stable/google-chrome-stable_116.0.5845.187-1_amd64.deb
RUN dpkg -i google-chrome-stable_116.0.5845.187-1_amd64.deb`
 
However, we don't want to use a fixed version.
We also have a task to check in future with the latest stable version to not remain behind with the versions

Thanks a lot for any suggestion

**Comment by portyu9** [comment]
Hi @francescoleddafrontiersin 

I would avoid writing logic to discover the newest Chrome `.deb` filename yourself.

If you genuinely want "whatever Chrome stable is current today", install through Google's package/repository mechanism so package resolution handles the version:

```dockerfile
RUN apt-get update \
    && apt-get install -y google-chrome-stable \
    && rm -rf /var/lib/apt/lists/*
```

after adding Google's Chrome repository/key to the image.

However, for CI the more reliable solution is actually the opposite: **pin the browser environment**.

Cypress publishes official Docker images such as:

```text
cypress/browsers
cypress/included
```

with tags that include exact Node/browser/Cypress versions. Cypress currently recommends the full frozen tags specifically because convenience tags such as `latest` and short-form tags can change when browsers are updated.

That gives you reproducible CI:

```dockerfile
FROM cypress/browsers:<full-node-and-browser-version-tag>
```

and you deliberately update that tag when you want to test a newer Chrome.

The failure you encountered is exactly the downside of consuming "latest" during every Docker build: a new browser release or packaging/propagation problem can break an otherwise unchanged pipeline.

So:

**Reproducible CI:** use a fully pinned `cypress/browsers` / `cypress/included` image and periodically update it.

**Always latest Chrome:** configure Google's APT repository and install `google-chrome-stable`; don't construct versioned `.deb` download URLs yourself.

Please mark this answer as accepted if it helped, thank you.
